model_hub/examples/huggingface/token-classification/ner_utils.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, List
+
+import datasets
+import numpy as np
+from attrdict import AttrDict
+from model_hub import huggingface as hf
+from seqeval import metrics
+
+
+def compute_metrics(label_list: List[Any], predictions: np.ndarray, labels: np.ndarray) -> Dict:
+    predictions = np.argmax(predictions, axis=2)
+
+    # Remove ignored index (special tokens)
+    true_predictions = [
+        [label_list[pr] for (pr, la) in zip(prediction, label) if la != -100]
+        for prediction, label in zip(predictions, labels)
+    ]
+    true_labels = [
+        [label_list[la] for (pr, la) in zip(prediction, label) if la != -100]
+        for prediction, label in zip(predictions, labels)
+    ]
+
+    return {
+        "accuracy_score": metrics.accuracy_score(true_labels, true_predictions),
+        "precision": metrics.precision_score(true_labels, true_predictions),
+        "recall": metrics.recall_score(true_labels, true_predictions),
+        "f1": metrics.f1_score(true_labels, true_predictions),
+    }
+
+
+def get_dataset_metadata(raw_datasets, hparams):
+    column_names = raw_datasets["train"].column_names
+    features = raw_datasets["train"].features
+    text_column_name = "tokens" if "tokens" in column_names else column_names[0]
+    label_column_name = (
+        "{}_tags".format(hparams.finetuning_task)
+        if "{}_tags".format(hparams.finetuning_task) in column_names
+        else column_names[1]
+    )
+
+    # Setup labels
+    if isinstance(features[label_column_name].feature, datasets.ClassLabel):
+        label_list = features[label_column_name].feature.names
+        # No need to convert the labels since they are already ints.
+        label_to_id = {i: i for i in range(len(label_list))}
+    else:
+        label_list = hf.get_label_list(raw_datasets["train"][label_column_name])
+        label_to_id = {l: i for i, l in enumerate(label_list)}
+
+    return AttrDict(
+        {
+            "num_labels": len(label_list),
+            "label_list": label_list,
+            "text_column_name": text_column_name,
+            "label_column_name": label_column_name,
+            "label_to_id": label_to_id,
+        }
+    )
+
+
+def build_tokenized_datasets(
```

Comment:
Nit: This is only going to load a single dataset at a time for a particular task, so should be change the function name `build_tokenized_datasets` and the parameter `raw_datasets` to be singular with respect to dataset?

Reply:
the returned datasets object has both a training and validation dataset, so that's why it's plural.  the original example also uses plural for the output from load_dataset:
https://github.com/huggingface/transformers/blob/master/examples/token-classification/run_ner.py#L238